### PR TITLE
Handle seed corpus execution in the replayer

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

File: tools/replayer/src/replayer.c [hunk 1]
```diff
@@ -447,9 +458,39 @@ int main(int argc, char **argv) {
...
+      assert(seed_corpus_path != NULL);
+#ifdef _WIN32
+      /* Use the non-deprecated safe versions of the standard string functions with the Microsoft CRT.
+       * Also, strip the ".exe" suffix to derive a platform-independent basename for the seed corpus. */
+      strncpy_s(seed_corpus_path, seed_corpus_path_size, argv[0], strlen(argv[0]) - strlen(".exe"));
+      strcat_s(seed_corpus_path, seed_corpus_path_size, SEED_CORPUS_SUFFIX);
+#else
+      /* sprintf_s is not available in Unix C90 system headers. */
+      sprintf(seed_corpus_path, "%s%s", argv[0], SEED_CORPUS_SUFFIX);
+#endif
```
Comment (adombeck) [inline tools/replayer/src/replayer.c:486]: Why do we try both `WITH_DEFAULT(cifuzz_seed_corpus)()` and `argv[0] + SEED_CORPUS_SUFFIX` here? Which cases are handled by each of these?

File: tools/replayer/src/replayer.c [hunk 2]
```diff
@@ -447,9 +458,39 @@ int main(int argc, char **argv) {
 
   WITH_DEFAULT(LLVMFuzzerInitialize)(&argc, &argv);
 
-  fprintf(stderr, "Running: <empty input>\n");
-  run_one_input(&empty[0], 0);
-  fprintf(stderr, "Done:    <empty input>: (0 bytes)\n");
+  /* If no arguments are specified, run the empty input and the seed corpus at argv[0] + SEED_CORPUS_SUFFIX
+   * Note: On Windows, ".exe" is stripped from argv[0] before forming the seed corpus path. */
+  if (argc == 1) {
```
Comment (adombeck) [inline tools/replayer/src/replayer.c:463]: Above a comment says that "If argv[0] + SEED_CORPUS_SUFFIX exists as a file or directory, its contents will always be executed first", but we only use if no arguments are specified

File: tools/replayer/src/replayer.c [hunk 3]
```diff
@@ -447,9 +458,39 @@ int main(int argc, char **argv) {
...
+    /* Always run the empty input. */
```
Comment (adombeck) [inline tools/replayer/src/replayer.c:464]: This comment seems misleading to me, this is only executed if no arguments are specified

File: tools/cmake/CIFuzz/share/CIFuzz/CIFuzzFunctions.cmake [hunk 4]
```diff
@@ -60,11 +60,19 @@ function(add_fuzz_test name)
...
+  set(_source_seed_corpus "${CMAKE_CURRENT_SOURCE_DIR}/${name}${_seed_corpus_suffix}")
+  # Convert path separators to '\' (Windows only) and escape all backslashes for a C string literal.
+  # In the regex strings below, one level of escaping is for the CMake string and another one to get a literal backslash
+  # in a regex.
+  if(WIN32)
+    string(REGEX REPLACE "/" "\\\\" _source_seed_corpus "${_source_seed_corpus}")
   endif()
+  string(REGEX REPLACE "\\\\" "\\\\\\\\" _source_seed_corpus "${_source_seed_corpus}")
+  # Compile the path to the seed corpus, which lives under the soure root, into the fuzz test binary as it is built
+  # out-of-tree and Windows doesn't have symlinks (junctions exist, but may cause issues with tools that are unaware of
```
[the remark this answers is not in the document]
Comment (fmeum) [reply tools/cmake/CIFuzz/share/CIFuzz/CIFuzzFunctions.cmake]: I reworked the comment to be more clear.

[the remark this answers is not in the document]
Comment (adombeck) [reply tools/cmake/CIFuzz/share/CIFuzz/CIFuzzFunctions.cmake]: Much better, thanks!

**Pull request merged**